### PR TITLE
0.4.0 - IDB Queue + Feats

### PR DESCRIPTION
**Pull request by kheina**
## Changelog
- feat: Use Indexed DB as queue backend for performant large queues
  - seriously, import the whole 200k+ full twitter blue verified list
- feat: add integration api
- feat: check users loaded on following/followers/subscriptions/verified followers/followers you know pages, see #120 

## Todo
- add limit to history db query
- add search bar to history

### CODE REVIEW

File: src/pages/queue/index.html [hunk 1]
```diff
@@ -10,31 +10,33 @@
...
 	<body>
 		<main>
-			<div class="inner">
-				<h1>Block Queue</h1>
-				<p class="subtitle">Users will not be added to the queue or blocked while this page is open.&nbsp;&nbsp;&nbsp;<a id="import-button">import <span id="import-arrow">▾</span></a></p>
-				<div id="importer">
-					<p>Blocklist import currently only support json files. files must be arrays containing the following type:</p>
+			<h1>Block Queue</h1>
+			<p class="subtitle">Users may be blocked while this page is open, refresh to update.&nbsp;&nbsp;&nbsp;<a id="import-button">import <span id="import-arrow">▾</span></a></p>
+			<p id="db-limit-reached" class="subtitle">holy shit your queue is huge, results have been limited to the first 10,000</p>
```
Comment (cooljeanius) [inline src/pages/queue/index.html:15]: is it expected that this could actually be reached?

**Review by cooljeanius — DISMISSED** [review]

**Comment by rougetimelord** [comment]
Went through and resolved merge conflicts as best as I can! Hopefully I didn't mess anything up

**Review by cooljeanius — DISMISSED** [review]

File: src/pages/queue/index.html (hunk 1 above)
Comment (cooljeanius) [reply src/pages/queue/index.html:15]: @rougetimelord WDYT?

**Review by rougetimelord — APPROVED** [review]

**Comment by rougetimelord** [comment]
I'll test then merge tomorrow!

File: src/pages/queue/index.html (hunk 1 above)
Comment (cooljeanius) [reply src/pages/queue/index.html:15]: ok, going to resolve this, then

**Comment by rougetimelord** [comment]
Ok that's all of the problems my JS linter found

**Review by cooljeanius — DISMISSED** [review]

File: src/background/index.ts [hunk 2]
```diff
@@ -1,7 +1,6 @@
-import { api, logstr, AddToHistoryAction, ErrorStatus, IsVerifiedAction, ReasonExternal, RemoveFromHistoryAction, SoupcanExtensionId, SuccessStatus, DefaultOptions } from '../constants';
+import { api, logstr, AddToHistoryAction, ErrorStatus, IsVerifiedAction, ReasonExternal, RemoveFromHistoryAction, SoupcanExtensionId, SuccessStatus, DefaultOptions, AddToQueueAction, PopFromQueueAction, IntegrationStateSendAndReceive, IntegrationStateDisabled, IntegrationStateSendOnly } from '../constants';
```
[the remark this answers is not in the document]
Comment (rougetimelord) [reply src/background/index.ts:1]: Disregard, figured it out :)

**Review by rougetimelord — APPROVED** [review]
Fixed it, all good. On to testing then merging #120 into this PR

**Comment by rougetimelord** [comment]
Alright, no more new code. Haven't run into any issues in my testing, but would still like a once over before merging

File: src/constants.ts [hunk 3]
```diff
@@ -105,3 +107,8 @@ export const ErrorStatus: ErrorStatus = 'ERROR';
 export const EventKey = 'MultiTabEvent';
 export const ErrorEvent = 'ErrorEvent';
 export const MessageEvent = 'MessageEvent';
+
+export const IntegrationStateDisabled = 0;
+export const IntegrationStateReceiveOnly= 1;
+export const IntegrationStateSendAndReceive = 2;
+export const IntegrationStateSendOnly = 3;
```
Comment (cooljeanius) [inline src/constants.ts:114]: fix up whitespace here please (newline at EOF)

**Review by cooljeanius — CHANGES_REQUESTED** [review]

File: src/constants.ts (hunk 3 above)
Comment (rougetimelord) [reply src/constants.ts]: Done!

**Review by cooljeanius — APPROVED** [review]

**Pull request merged**